pkg/utils/utils.go

```diff
@@ -77,6 +80,13 @@ func Hash(value string, hashLength int) string {
 // This function shortens a name to the maximum length given and uses rest of the string that is too long
 // as hash that gets added to the valid name.
 func ShortenedName(name string, maxLength, hashLength int) string {
+	actualBound := int(math.Min(float64(maxLength), float64(len(name))))
+	// First shorten so if there is a problem it's due to ilegal chars, rahter its legnth
+	if ilegal := validation.IsDNS1123Subdomain(name[:actualBound]); ilegal != nil {
```

Comment:
I don't understand the check. The function returns a string array which is never nil, so you will always do hashing even if the name is ok.
`func IsDNS1123Subdomain(value string) []string {`